examples/lerax-square.cpp

```diff
@@ -53,75 +67,35 @@ struct Player {
     }
 };
 
-class Game {
+
+class Game: public GameLoop {
 private:
-    SDL_Window *window = nullptr;
-    SDL_Renderer *renderer = nullptr;
-    const char *title = "Square Moving";
-    bool running = true;
-    struct Player player;
+    using GameLoop::GameLoop;
+    struct Player player = {};
     struct KeyboardState keyboard = {};
-
-    void event(SDL_Event *e) {
-        while (SDL_PollEvent(e)) {
-            switch (e->type) {
-            case SDL_QUIT:
-                running = false;
-                break;
-            case SDL_KEYDOWN:
-                switch(e->key.keysym.sym) {
-                case SDLK_ESCAPE:
-                    running = false;
-                    break;
-                case SDLK_UP:
-                    keyboard.up = true;
-                    break;
-                case SDLK_DOWN:
-                    keyboard.down = true;
-                    break;
-                case SDLK_LEFT:
-                    keyboard.left = true;
-                    break;
-                case SDLK_RIGHT:
-                    keyboard.right = true;
-                    break;
-                case SDLK_w:
-                    keyboard.velocity_up = true;
-                    break;
-                case SDLK_s:
-                    keyboard.velocity_down = true;
-                    break;
-                }
-                break;
-            case SDL_KEYUP:
-                switch(e->key.keysym.sym) {
-                case SDLK_UP:
-                    keyboard.up = false;
-                    break;
-                case SDLK_DOWN:
-                    keyboard.down = false;
-                    break;
-                case SDLK_LEFT:
-                    keyboard.left = false;
-                    break;
-                case SDLK_RIGHT:
-                    keyboard.right = false;
-                    break;
-                }
-                break;
-            }
-        }
+    InputHandler input_handler = InputHandler(input_config);
+
+    void event() {
+        static SDL_Event e;
+        input_handler.process(e);
+        keyboard.up = input_handler.read("up");
```

Comment:
I like this! But would the semantic meaning of this function would be better read as `act` instead of `read`?

Reply:
Why act?

Reply:
Because the input handler would act over the "up" event instead of reading the event (?).

Reply:
Actually it's just read the state map of a given key

Reply:
I think the intention wasn't propagate events but only get a key, sure?

Reply:
Yes, maybe using `get` and `set` can be more appropriate in this case.